### PR TITLE
Use exec state

### PR DESCRIPTION
**Pull request by cryptoAtwill**
This addresses issue #176.

This PR removes the state_tree option and just use exec_state, which also exposes a state_tree_mut() method now, to avoid any discrepancy, and then you can chain call and actor_state calls freely.

### CODE REVIEW

File: fendermint/vm/interpreter/src/fvm/state/query.rs
Comment (adlrocha) [inline fendermint/vm/interpreter/src/fvm/state/query.rs:111]: I will probably let @aakoshh have the final call in this one because I may be missing something, so let's wait for his review, but do we really need `use_cache` as a public argument of `actor_state`? Isn't `use_cache` always `false` in this case? 
```suggestion
        self.with_exec_state(false, |exec_state| {
```

**Review by adlrocha — APPROVED** [review]
Thanks! Let's wait for @aakoshh review before merging.

File: fendermint/vm/interpreter/src/fvm/state/query.rs
Comment (aakoshh) [reply fendermint/vm/interpreter/src/fvm/state/query.rs:111]: I think it makes sense to offer it, the calls can now influence the result, and you might want to query it taking the last call into account.

File: fendermint/vm/interpreter/src/fvm/state/query.rs
Comment (aakoshh) [inline fendermint/vm/interpreter/src/fvm/state/query.rs:140]: It this was moved inside the with_exec_state then it wouldn’t have to initialise the execution state twice.

**Review by aakoshh — APPROVED** [review]
Nice, thanks for picking this up Will!

File: fendermint/vm/interpreter/src/fvm/state/query.rs
Comment (cryptoAtwill) [reply fendermint/vm/interpreter/src/fvm/state/query.rs]: yeah, @aakoshh, I removed the `actor_state` call and directly used the `exec_state`.

**Pull request merged**
